### PR TITLE
Feat(mysql): convert full outer join to union with left/right outer joins

### LINKED ISSUE #2458: Feat(mysql): transpile FULL OUTER JOIN to LEFT/UNION/RIGHT
MySQL does not support `FULL OUTER JOIN`. Whenever sqlglot is transpiling from another dialect, I'd like it to generate the equivalent operation of `LEFT JOIN` `UNION` `RIGHT JOIN`.

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #2458

This is an incomplete fix since the generic solution is more tricky.

### CODE REVIEW

File: sqlglot/transforms.py [hunk 1]
```diff
@@ -374,6 +374,30 @@ def eliminate_semi_and_anti_joins(expression: exp.Expression) -> exp.Expression:
...
+            if join.side == "FULL" and join.kind == "OUTER"
+        ]
+
+        if len(full_outer_joins) == 1:
+            expression_copy = expression.copy()
+            index, full_outer_join = full_outer_joins[0]
+            full_outer_join.set("side", "left")
+            expression_copy.args["joins"][index].set("side", "right")
+
+            return exp.union(expression, expression_copy)
```
Comment (tobymao) [inline sqlglot/transforms.py:396]: copy=False here?

**Review by tobymao — APPROVED** [review]

File: sqlglot/transforms.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/transforms.py:396]: `exp.union` doesn't copy by default it seems, should I change this?

Comment (tobymao) [reply sqlglot/transforms.py:396]: yea

Comment (GeorgeSittas) [reply sqlglot/transforms.py]: done

**Pull request merged**